kork-core/src/main/java/com/netflix/spinnaker/kork/spring/PluginLoader.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2019 Armory, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.netflix.spinnaker.kork.spring;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+public class PluginLoader {
+
+  static final String DEFAULT_PLUGIN_CONFIG_PATH = "/opt/spinnaker/config/plugins.yml";
+
+  private String filename;
+  private List<URL> urls;
+
+  PluginLoader(String filename) {
+    this.filename = filename;
+  }
+
+  PluginLoader() {
+    this.filename =
+        Optional.ofNullable(System.getenv("PLUGIN_CONFIG_LOCATION"))
+            .orElse(PluginLoader.DEFAULT_PLUGIN_CONFIG_PATH);
+  }
+
+  public void loadPlugins(Class source) throws PluginLoaderException {
+    try (InputStream inputStream = openFromFile()) {
+      PluginProperties props = parsePluginConfigs(inputStream);
+      this.urls = getEnabledJarURLs(props);
+    } catch (Exception e) {
+      throw new PluginLoaderException(e);
+    }
+    addJarsToClassPath(source);
+  }
+
+  private PluginProperties parsePluginConfigs(InputStream inputStream)
+      throws PluginLoaderException {
+    ObjectMapper objectMapper = new ObjectMapper(new YAMLFactory());
+    try {
+      PluginProperties pluginConfigs = objectMapper.readValue(inputStream, PluginProperties.class);
+      return pluginConfigs;
+    } catch (IOException e) {
+      throw new PluginLoaderException("Unable to parse plugin configurations", e);
+    }
+  }
+
+  private InputStream openFromFile() throws IOException {
+    return new FileInputStream(this.filename);
+  }
+
+  private List<URL> getEnabledJarURLs(PluginProperties pluginProperties)
+      throws PluginLoaderException {
+    try {
+      return pluginProperties.pluginConfigurationList.stream()
+          .filter(isEnabled())
+          .map(PluginProperties.PluginConfiguration::getPath)
+          .map(Paths::get)
+          .map(this::getUrlfromPath)
+          .collect(Collectors.toList());
+    } catch (Exception e) {
+      throw new PluginLoaderException("Unable to get plugin jar paths", e);
```

Comment:
Nitpick but `Unable to get plugin jar paths` => `Unable to get plugin paths` or `Unable to get plugin JAR paths`